lib/queries.ts

```diff
@@ -7,6 +7,7 @@ export const entriesQuery = `*[_type == "entry"]{...,  mainImage {..., file {...
 export const entryBySlugQuery = `*[slug.current == $slug]`
 export const patternsQuery = `*[_type == "pattern"]`
 export const patternClassesQuery = `*[_type == "patternClass"] | order(order)`
+export const patternsWithClassQuery = `*[_type == "pattern"]{..., class -> }`
```

Comment:
```suggestion
export const patternsWithClassQuery = `
  *[_type == "pattern"]
    { ..., class -> }
    [ count(*[_type == "entry" && references(^._id)]) > 0]
`
```
We could narrow this down further by only showing pattern which are associated to entries.